src/Http/HeaderParser.php

```diff
@@ -0,0 +1,128 @@
+<?php
+declare(strict_types=1);
+
+namespace Cake\Http;
+
+/**
+ * Provides methods which parse HTTP header strings
+ */
+class HeaderParser
```

Comment:
Perhaps we can use the suffix `Utils` or similar instead of `Parser` and not narrow down what this class can do. That would potentially allow us to put more header related stuff in here if required. For e.g. https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/HttpFoundation/HeaderUtils.php